public/app/features/logs/logsModel.ts

```diff
@@ -274,18 +275,21 @@ export function getSeriesProperties(
   let requestedRangeMs;
   // Clamp time range to visible logs otherwise big parts of the graph might look empty
   if (absoluteRange) {
-    const earliestTsLogs = sortedRows[0].timeEpochMs;
+    const firstTimeStamp = sortedRows[0].timeEpochMs;
+    const lastTimeStamp = sortedRows[sortedRows.length - 1].timeEpochMs;
+    const earliestTsLogs = firstTimeStamp < lastTimeStamp ? firstTimeStamp : lastTimeStamp;
+    const earliestLogToTimeRangeEnd = absoluteRange.to - earliestTsLogs;
 
     requestedRangeMs = absoluteRange.to - absoluteRange.from;
-    visibleRangeMs = absoluteRange.to - earliestTsLogs;
+    visibleRangeMs = Math.abs(firstTimeStamp - lastTimeStamp);
```

Comment:
Do we want to force this to a positive number always the next line was checking if it was greater than zero?

Reply:
Another good question, this method was probably written before we had support for forward log search, so the assumption that it was always positive made more sense as a gate to ensure the data was as expected, but with supporting sorts in both directions we don't really care if it's positive or negative, we just want to render the duration that the logs span